pkgs/build-support/cc-wrapper/setup-hook.sh

```diff
@@ -147,6 +147,10 @@ export ${role_pre}CXX=@named_cxx@
 export CC${role_post}=@named_cc@
 export CXX${role_post}=@named_cxx@
 
+# If unset, assume the default hardening flags.
+: ${NIX_HARDENING_ENABLE="fortify stackprotector pic strictoverflow format relro bindnow"}
```

Comment:
When I split up into two variables, this will contain just the flags that make sense for the cc wrapper. Similarly for the bintools' setup-hook.

Reply:
(For posterity: per IRC convo, I not sure if we're going with the split variables over the unified variable approach. In light of this, the above comment might be misleading.)